core/snippets/snippets_insert_raw_text.py

```diff
@@ -122,7 +122,9 @@ def is_any_line_from_right_to_left(lines) -> bool:
                 next_key = next_stop.compute_sorting_key()
                 # If the ordering between the keys and columns are inconsistent,
                 # the stops on this line go from right to left
-                if (next_key < stop_key) != (stop.col < next_stop.col):
+                is_next_key_smaller = next_key < stop_key
+                is_next_col_smaller = next_stop.col < stop.col
+                if is_next_key_smaller != is_next_col_smaller:
```

Comment:
```suggestion
                if (next_key < stop_key) != (next_stop.col < stop.col):
```
The logic was just backwards.

Reply:
Yes, because Python's operator precedence isn't what you think it is; [all comparison operators are at the same precedence level](https://docs.python.org/3/reference/expressions.html#comparisons).